pkg/operator/sync_configmapcabundle_v311_00.go

```diff
@@ -0,0 +1,130 @@
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsclientv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	operatorsv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	scsv1alpha1 "github.com/openshift/api/servicecertsigner/v1alpha1"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcecread"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/service-serving-cert-signer/pkg/operator/v310_00_assets"
+)
+
+// syncConfigMapCABundleController_v311_00_to_latest takes care of synchronizing (not upgrading) the thing we're managing.
+// most of the time the sync method will be good for a large span of minor versions
+func syncConfigMapCABundleController_v311_00_to_latest(c ServiceCertSignerOperator, operatorConfig *scsv1alpha1.ServiceCertSignerOperatorConfig, previousAvailability *operatorsv1alpha1.VersionAvailablity) (operatorsv1alpha1.VersionAvailablity, []error) {
+	versionAvailability := operatorsv1alpha1.VersionAvailablity{
+		Version: operatorConfig.Spec.Version,
+	}
+
+	errors := []error{}
+	var err error
+
+	requiredNamespace := resourceread.ReadNamespaceV1OrDie(v310_00_assets.MustAsset("v3.10.0/configmap-cabundle-controller/ns.yaml"))
+	if _, _, err = resourceapply.ApplyNamespace(c.corev1Client, requiredNamespace); err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "ns", err))
+	}
+
+	requiredClusterRole := resourceread.ReadClusterRoleV1OrDie(v310_00_assets.MustAsset("v3.10.0/configmap-cabundle-controller/clusterrole.yaml"))
+	if _, _, err = resourceapply.ApplyClusterRole(c.rbacv1Client, requiredClusterRole); err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "svc", err))
+	}
+
+	requiredClusterRoleBinding := resourceread.ReadClusterRoleBindingV1OrDie(v310_00_assets.MustAsset("v3.10.0/configmap-cabundle-controller/clusterrolebinding.yaml"))
+	if _, _, err = resourceapply.ApplyClusterRoleBinding(c.rbacv1Client, requiredClusterRoleBinding); err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "svc", err))
+	}
+
+	requiredService := resourceread.ReadServiceV1OrDie(v310_00_assets.MustAsset("v3.10.0/configmap-cabundle-controller/svc.yaml"))
+	if _, _, err = resourceapply.ApplyService(c.corev1Client, requiredService); err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "svc", err))
+	}
+
+	requiredSA := resourceread.ReadServiceAccountV1OrDie(v310_00_assets.MustAsset("v3.10.0/configmap-cabundle-controller/sa.yaml"))
+	_, saModified, err := resourceapply.ApplyServiceAccount(c.corev1Client, requiredSA)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "sa", err))
+	}
+
+	// TODO create a new configmap whenever the data value changes
+	_, configMapModified, err := manageConfigMapCABundleConfigMap_v311_00_to_latest(c.corev1Client, operatorConfig)
```

Comment:
This should be pulled out of the api into a CRD - there is no reason it needs to be in the api and now is the time to pull this stuff out.